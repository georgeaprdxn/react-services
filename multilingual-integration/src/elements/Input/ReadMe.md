Input

```jsx inside Markdown
const [value, setVal] = React.useState('Some text')
const changeHandler = e => setVal(e.target.value)
;<Input row={3} val={value} handleChange={e => changeHandler(e)} />
```
