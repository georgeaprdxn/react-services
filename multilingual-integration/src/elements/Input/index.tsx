import React, { FunctionComponent } from 'react'

import { StyledInput } from './styled'

type Props = {
	/**
	 * type of input
	 */
	inputType?: string
	/**
	 * value of input
	 */
	val: string
	/**
	 * handles change event
	 */
	handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const Input: FunctionComponent<Props> = ({ inputType, val, handleChange }) => {
	return (
		<StyledInput
			type={inputType}
			value={val}
			onChange={handleChange}
			className="custom-input"
		/>
	)
}

Input.defaultProps = {
	inputType: 'text',
	handleChange: () => undefined,
}

export default Input
