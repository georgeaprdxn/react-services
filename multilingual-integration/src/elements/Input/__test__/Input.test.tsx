import React from 'react'
import { shallow } from 'enzyme'

import Input from '../index'
import mockInputData from './Input.mock.json'

describe('Input element tests', () => {
	let inputElem: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		inputElem = shallow(
			<Input
				inputType={mockInputData.inputType}
				val={mockInputData.val}
				handleChange={handleChangeStub}
			/>
		)
	})

	it('should render a <input />', () => {
		expect(inputElem.find('.custom-input').length).toEqual(1)
	})

	it('validates all props', () => {
		expect(inputElem.find('.custom-input').prop('value')).toEqual(
			mockInputData.val
		)
		expect(inputElem.find('.custom-input').prop('type')).toEqual(
			mockInputData.inputType
		)
	})

	it('checks the DOM events', () => {
		inputElem
			.find(`.custom-input`)
			.simulate('change', { target: { value: mockInputData.val } })

		expect(handleChangeStub).toHaveBeenCalledTimes(1)
	})

	it('matches the snaphot', () => {
		expect(inputElem).toMatchSnapshot()
	})
})
