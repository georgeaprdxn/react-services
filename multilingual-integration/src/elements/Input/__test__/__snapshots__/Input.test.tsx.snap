// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input element tests matches the snaphot 1`] = `
<styled.input
  className="custom-input"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "target": Object {
              "value": "Some text",
            },
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  type="text"
  value="Some text"
/>
`;
